### ISSUE TITLE
Leaflet / openstreet map links only identified by colour

### ISSUE THREAD

**Issue by minorOffense** [Leaflet/Leaflet#6742]
The maps have 'leaflet' link which is identified only by color.

Criteria 1.4.1
Compliance success level A

![0f49f0d4-0e1d-11e7-8e12-1a83beefc03c](https://user-images.githubusercontent.com/871728/62141892-52d79780-b2bb-11e9-925d-2046f703f4cb.png)

- [x] I've looked at the [documentation](http://leafletjs.com/reference.html) to make sure the behavior is documented and expected
- [x] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Cordova, Ionic, Angular, React…)
- [x] I've searched through the issues to make sure it's not yet reported

**Steps to reproduce**
- View a map
- Look at the link text in the bottom right
- No visual indication that text is a link other than the colour difference

**Expected behavior**
When the link has focus or hover it should show an underline or some other visual indicator to pass WCAG compliance.
See https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html

**Current behavior**
Only identified by a colour.
See https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/F73 for an example failure.

**Environment**
- Leaflet version: all
- Browser (with version): all
- OS/Platform (with version): all

**Comment by Malvoz** [comment]
[Success Criterion 1.4.1 Use of Color](https://www.w3.org/TR/WCAG21/#use-of-color):

> Color is not used as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element.

There _is_ `text-decoration` on `:hover`, so the only issue is that when links/buttons receive focus the outlines aren't persistent, that problem is mainly covered in https://github.com/Leaflet/Leaflet/issues/6986, I think this can be closed in favor of that issue.

P.S. @minorOffense you're referring to WCAG 2.0 documentation, latest recommendation is [2.1](https://www.w3.org/TR/WCAG21/).

**Comment by Malvoz** [comment]
> See https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html

The latest published version is https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html (WCAG 2.2 is around the corner).

> No visual indication that text is a link other than the colour difference

As long as the difference in lightness (contrast) is 3:1 or greater with the surrounding text they still pass <span title="Success Criterion">SC</span> [1.4.1 Use of Color](https://www.w3.org/TR/WCAG21/#use-of-color). That contrast ratio is currently 2.56.

> **Expected behavior**
> When the link has focus or hover it should show an underline

WCAG 2.1 [Technique F73](https://www.w3.org/WAI/WCAG21/Techniques/failures/F73.html) states:
<q>If the non-color cue only happens when the mouse hovers over the link or when the link receives focus, it is still a failure.</q>
so that wouldn't be sufficient, it'd need to always have an underline.

It also states: <q>If the link is a different color and bold it would not fail because the boldness is not color dependent</q>.

<hr>

In other words, for links not to fail they should meet at least 1 of the following:
- Preserve the default `text-decoration` underline (in the default state, not only on hover/focus)
- Use `font-weight: bold`
- Have a contrast ratio of 3:1 with the surrounding text

OTOH, attribution can still be hard to see due to the applied [alpha transparency on the attribution container](https://github.com/Leaflet/Leaflet/blob/0f904a515879fcd08f69b7f51799ee7f18f23fd8/dist/leaflet.css#L405), depending on the underlying graphic. But that's a separate issue.

**Comment by obfuscatedgenerated** [comment]
I've also had [axe-core alerting me to this](https://dequeuniversity.com/rules/axe/4.1/link-in-text-block). In my case, I just used a shade of blue with more contrast, but I think the underline or weighting could also be acceptable. In my personal opinion, the bold looks the best out of those options, but it might lead to unexpected new text wrapping for people putting longer content in there, e.g:

Normal:
<img width="552" height="40" alt="Image" src="https://github.com/user-attachments/assets/e456ac56-2801-4e1c-a176-937a395bbe27" />

Bolder:
<img width="553" height="54" alt="Image" src="https://github.com/user-attachments/assets/4c6b36c4-4d1c-4a71-9066-75d0788b3301" />
